fhir-persistence-schema/docs/SchemaDesign.md

```diff
@@ -0,0 +1,650 @@
+
+The IBM FHIR Server uses soft-delete when processing a FHIR `DELETE` interaction. This creates a new version of the resource with a minimal resource data value and the `IS_DELETED` flag = `Y`. Soft-deletes do not delete existing database records. The Patient Erase custom operation can be used to remove all traces of a patient from a database.
+
+All timestamps stored in the IBM FHIR Server schema are UTC.
+
+For more details on how the resource payload data is stored, see the next section.
+
+## Scanning Resources
+
+The IBM FHIR Server implements the whole-system `_history` endpoint to fetch resources in the order they were ingested into the system. This endpoint is described detail in the IBM FHIR Server [Conformance Guide](https://ibm.github.io/FHIR/Conformance/#whole-system-history) guide. This service is backed by the `RESOURCE_CHANGE_LOG` table which records the identity of each resource version as it is ingested. The table is indexed on `CHANGE_TSTAMP` which reflects the UTC last-modified timestamp of the version of the resource.
+
+```
+fhirdb=> \d fhirdata.resource_change_log
+                        Table "fhirdata.resource_change_log"
+       Column        |            Type             | Collation | Nullable | Default 
+---------------------+-----------------------------+-----------+----------+---------
+ resource_id         | bigint                      |           | not null | 
+ resource_type_id    | integer                     |           | not null | 
+ logical_resource_id | bigint                      |           | not null | 
+ change_tstamp       | timestamp without time zone |           | not null | 
+ version_id          | integer                     |           | not null | 
+ change_type         | character(1)                |           | not null | 
+Indexes:
+    "resource_change_log_pk" PRIMARY KEY, btree (resource_id)
+    "unq_resource_change_log_ctrtri" UNIQUE, btree (change_tstamp, resource_type_id, resource_id)
+```
+
+As reflected in the whole-system-history REST API, the `RESOURCE_CHANGE_LOG` table can be scanned in two ways:
+
+1. Ordered by `RESOURCE_ID`, following the natural order of the primary key
+2. Ordered by `CHANGE_TSTAMP`, following the timeline of resources as they arrive
+
+Filtering and ordering based on `RESOURCE_ID` is the simplest approach because there are no duplicates to deal with although care is still required when reading data near the current time (within the window of the maximum transaction timeout time). This is because ids are allocated before the transaction commits. It is therefore possible for records with a smaler `RESOURCE_ID` value to appear after records which have already been committed. This is not a limitation of the IBM FHIR Server, but just a side-effect that is common in systems like this.
```

Comment:
```suggestion
Filtering and ordering based on `RESOURCE_ID` is the simplest approach because there are no duplicates to deal with although care is still required when reading data near the current time (within the window of the maximum transaction timeout time). This is because ids are allocated before the transaction commits. It is therefore possible for records with a smaller `RESOURCE_ID` value to appear after records which have already been committed. This is not a limitation of the IBM FHIR Server, but just a side-effect that is common in systems like this.
```

Reply:
ok